Tests/SonarScanner.MSBuild.PreProcessor.Test/RoslynRuleSetGeneratorTests.cs

```diff
@@ -225,51 +194,73 @@ public void RoslynRuleSet_Common_Parameters()
     [TestMethod]
     public void RoslynRuleSet_AnalyzerId_Proprety_Missing()
     {
-        var generator = new RoslynRuleSetGenerator(new ListPropertiesProvider(new Dictionary<string, string>
-        {
-            { "sonaranalyzer-cs.ruleNamespace", "SonarAnalyzer.CSharp" },
-        }));
-        var rules = new[]
-        {
-            new SonarRule("csharpsquid", "active1", true),
-        };
-
-        new Action(() => generator.Generate("cs", rules)).Should()
+        var context = new RuleSetTestContext(
+            [new("csharpsquid", "active1", true)],
+            new Dictionary<string, string>
+            {
+                { "sonaranalyzer-cs.ruleNamespace", "SonarAnalyzer.CSharp" },
+            });
```

Comment:
You can make these more concise, there's so much sugar that the more verbose code doesn't help anyway
```suggestion
            new() { { "sonaranalyzer-cs.analyzerId", "SonarAnalyzer.CSharp" } });
```